### DISCUSSION TITLE
Build instructions for linux

### DISCUSSION THREAD

**Discussion by johannesjo** [AppFlowy-IO/AppFlowy#30] [Technical help wanted] [answered]
I am one of the main developers of the open source app [super productivity](https://github.com/johannesjo/super-productivity). Since this is – very broadly speaking – in similar category (productivity & knowledge organization) I was super curious to check this out. Since I am working form Linux I am curious if there are any build instructions available for the platform.

**Comment by pdckxd** [comment, marked as answer]
@johannesjo , you can check this link https://github.com/AppFlowy-IO/appflowy/blob/main/doc/BUILD_ON_LINUX.md

**Reply by johannesjo** [reply]
Thank you very much! Will there be any pre built binaries at some point?

**Reply by pdckxd** [reply]
np, I believe @appflowy will publish Linux version soon.

**Reply by atrevino89** [reply]
Hey guys, this is awesome, congrats about the great product and tech stack used !
A quick question... wondering if Brew is used only to install Rust? asking this because I wouldn't like to install that just for AppFlowy's sake (I got already multiple package managers and want to be minimal on that front) 
Have a great one!

**Reply by pdckxd** [reply]
I think so. But I didn’t try. I am going to remove brew steps in instruction.

**Reply by ghost** [reply]
@thxv3n0lvl The part where brew is installed can be just replaced by the official install of rustup, which @pdckxd will do at some point

**Reply by pdckxd** [reply]
BUILD_ON_LINUX.md is updated. brew dep was removed. See: https://github.com/AppFlowy-IO/appflowy/blob/main/doc/BUILD_ON_LINUX.md

**Comment by lenaMK** [comment]
Hi, I just tried the [Linux install instructions](https://github.com/AppFlowy-IO/appflowy#linux)
At the last step(6: Run the app),  I get this error message: 
```
[ERROR:flutter/lib/ui/ui_dart_state.cc(209)] Unhandled Exception: Invalid argument(s): Failed to load dynamic library 'libdart_ffi.so': /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.32' not found (required by /opt/appflowy/AppFlowy/lib/libdart_ffi.so)

```
I think it might be because I don't have rust and flutter installed. 
Could you please confirm if that's the case and may I suggest adding these steps in the install instructions? 
Thank you, looking forward to trying Flowy!

**Reply by ghost** [reply]
Distro?

Can you do `ldd --version`? You need at least glibc 2.32.

> I think it might be because I don't have rust and flutter installed.

You don't need it, they're pre-compiled binaries.

**Reply by lenaMK** [reply]
Indeed, I'm on `ldd (Ubuntu GLIBC 2.31-0ubuntu9.2) 2.31`
I tried to install glibc 2.32 following [a forum chat](https://askubuntu.com/questions/1345342/how-to-install-glibc-2-32-when-i-already-have-glibc2-31), I guess I still need to show Flowy where glibc-2.32 is?

**Comment by Pada0** [comment]
an AMD64 linux version would be greatly appreciated
